projects/components/src/download-json/download-json.component.ts

```diff
@@ -0,0 +1,79 @@
+import { DOCUMENT } from '@angular/common';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Inject, Input, Renderer2 } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import { IconSize } from '@hypertrace/components';
+import { Observable } from 'rxjs';
+import { catchError, finalize, take } from 'rxjs/operators';
+import { ButtonSize, ButtonStyle } from '../button/button';
+import { NotificationService } from '../notification/notification.service';
+
+@Component({
+  selector: 'ht-download-json',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./download-json.component.scss'],
+  template: `
+    <div class="download-json" [htTooltip]="this.tooltip" (click)="this.triggerDownload()">
+      <ht-button
+        *ngIf="!this.dataLoading"
+        class="download-button"
+        icon="${IconType.Download}"
+        display="${ButtonStyle.Text}"
+        size="${ButtonSize.Large}"
+      ></ht-button>
+      <ht-icon *ngIf="this.dataLoading" icon="${IconType.Loading}" size="${IconSize.Large}"></ht-icon>
+    </div>
+  `
+})
+export class DownloadJsonComponent {
+  @Input()
+  public dataSource!: Observable<unknown>;
+
+  @Input()
+  public fileName: string = 'download';
+
+  @Input()
+  public tooltip: string = 'Download Json';
```

Comment:
Please remove tooltip as indicated on the other PR

Reply:
Would do in other PR